src/test/java/io/fabric8/maven/docker/config/BuildImageConfigurationTest.java

```diff
@@ -208,5 +209,19 @@ public void compression() {
         assertEquals(none, config.getCompression());
     }
 
+    @Test
+    public void multipleAssembliesUniqueNames() {
+        AssemblyConfiguration assemblyConfigurationOne = new AssemblyConfiguration.Builder().name("foo").build();
+        AssemblyConfiguration assemblyConfigurationTwo = new AssemblyConfiguration.Builder().name("foo").build();
+        BuildImageConfiguration config =
+                new BuildImageConfiguration.Builder().
+                        assemblies(Arrays.asList(assemblyConfigurationOne, assemblyConfigurationTwo)).build();
 
+        try {
+            config.initAndValidate(logger);
+        } catch (IllegalArgumentException expected) {
+            return;
+        }
+        fail("Should have failed.");
```

Comment:
We can use the expected attribute of the `@Test` annotation to declare `IllegalArgumentException` to be thrown anywhere in the annotated test method like this: `@Test(expected = IllegalArgumentException.class)`